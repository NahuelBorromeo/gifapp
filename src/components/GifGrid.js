import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    //El useEffect nos va a permitir ejecutar cierto codigo de manera condicional


    //usamos nuestro custom hook
    const { data:images, loading } = useFetchGifs( category );

    // useEffect( () => {
    //     getGifs( category )
    //         .then( setImages );
    // }, [ category ])
    


    return (
        <>
            <h3 className='animate__animated animate__fadeIn' >{ category }</h3>

            { loading && <p className='animate__animated animate__flash'>Loading</p> }

            <div className='card-grid'>
                    { 
                        images.map( img => 
                            <GifGridItem 
                                    key={ img.id }
                                    { ...img }
                            />)
                    }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
