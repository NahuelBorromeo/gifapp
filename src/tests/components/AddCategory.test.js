import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {
    //jest.nf() es lo mismo que poner una funci√≥n simple, solo que nos permite saber cuando y cuantas veces fue llamada.
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories } />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } />);
    });

    test('should to show correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('should to change the input', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', { target: { value } });

        expect( wrapper.find('p').text().trim() ).toBe( value );
    });
    
    test('should not to post the information on submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        
        expect( setCategories ).not.toHaveBeenCalled();

    })
    
    test('should to call the func setCategories and clear the textbox', () => {
        
        const value = 'Hola Mundo';
        
        wrapper.find('input').simulate('change', { target: { value }});

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        
        expect( wrapper.find('input').prop('value') ).toBe('');
    })
    
    

})
