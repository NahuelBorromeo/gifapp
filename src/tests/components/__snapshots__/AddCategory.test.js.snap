// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> should to show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
